plc4j/spi/src/main/java/org/apache/plc4x/java/spi/messages/DefaultPlcWriteRequest.java

```diff
@@ -194,6 +196,9 @@ public PlcWriteRequest build() {
                 PlcTag tag = tagValues.getLeft().get();
                 Object[] value = tagValues.getRight();
                 PlcValue plcValue = valueHandler.newPlcValue(tag, value);
+                if(tag.getPlcValueType()== PlcValueType.NULL){
```

Comment:
As far as I understood it PlcValueType.NULL is used in cases where a value can be null (Like in pointers etc.) ... so it seems as only if you set the type according to the plcValue, if this is NULL ... should not be confused with "null".

Reply:
By the way. The value can be set null not PlcValueType be set NULL. PLC4x driver write request must give explicit data type. This limits all the driver write tag format for the PLC4x

Reply:
You can checkout my repo and take look PL4x driver and PL4x server